prow/config/inrepoconfig.go

```diff
@@ -187,3 +188,46 @@ func DefaultAndValidateProwYAML(c *Config, p *ProwYAML, identifier string) error
 
 	return utilerrors.NewAggregate(errs)
 }
+
+type InRepoConfigGitCache struct {
+	git.ClientFactory
+	cache map[string]git.RepoClient
+	lock  sync.Mutex
+}
+
+func NewInRepoConfigGitCache(factory git.ClientFactory) git.ClientFactory {
+	return &InRepoConfigGitCache{
+		ClientFactory: factory,
+		cache:         map[string]git.RepoClient{},
+	}
+}
+
+func (c *InRepoConfigGitCache) ClientFor(org, repo string) (git.RepoClient, error) {
```

Comment:
Nit: `GetClientFor` would read easier for me; but at the same time I notice that it is an existing naming convention (e.g., `c.ClientFactory.ClientFor()` on line 217). Nvm

Reply:
Yeah, it also has to have that exact name to match this interface: https://pkg.go.dev/k8s.io/test-infra/prow/git/v2#ClientFactory